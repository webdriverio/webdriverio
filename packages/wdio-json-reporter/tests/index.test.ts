import { describe, expect, it, vi } from 'vitest'

import JSONReporter from '../src/index.js'

vi.mock('node:fs', () => ({
    default: {
        mkdirSync: vi.fn(),
        createWriteStream: vi.fn()
    }
}))

describe('JSONReporter', () => {
    it('should create a report', async () => {
        const reporter = new JSONReporter({
            logFile: '/foo/bar.log',
            stdout: true
        })
        expect(reporter.options.logFile).toBe('/foo/bar.json')
    })

    it('should prepare json', () => {
        const reporter = new JSONReporter({ stdout: true })
        reporter.write = vi.fn()
        reporter.suites = {
            '0-0': {
                title: 'foo',
                fullTitle: 'foo',
                state: 'passed',
                duration: 1,
                hooks: [{
                    title: 'before each',
                    fullTitle: 'foo "before each"',
                    state: 'failed',
                    errors: [new Error('ups')],
                    duration: 1
                }],
                tests: [{
                    title: 'foo',
                    fullTitle: 'foo',
                    state: 'passed',
                    duration: 1
                }],
            } as any,
            '0-1': {
                title: 'bar',
                fullTitle: 'bar',
                state: 'failed',
                duration: 2,
                error: new Error('ahh'),
                hooks: [{
                    title: 'before each',
                    fullTitle: 'foo "before each"',
                    state: 'passed',
                    duration: 1
                }],
                tests: [{
                    title: 'bar',
                    fullTitle: 'bar',
                    state: 'failed',
                    duration: 2,
                    error: new Error('ahh')
                }],
            } as any
        }
        reporter.onRunnerEnd({
            start: 0,
            end: 1,
            capabilities: {},
            config: {
                framework: 'mocha',
                mochaOpts: {}
            },
            specs: ['foo.js', 'bar.js']
        } as any)
        expect(vi.mocked(reporter.write).mock.calls).toMatchSnapshot()
    })
})
