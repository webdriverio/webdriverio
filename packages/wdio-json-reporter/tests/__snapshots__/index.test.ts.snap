// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JSONReporter > should prepare json 1`] = `
[
  [
    "{"start":0,"end":1,"capabilities":{},"framework":"mocha","mochaOpts":{},"suites":[{"name":"foo","tests":[{"name":"foo","duration":1,"state":"passed"}],"hooks":[{"duration":1,"title":"before each","state":"failed"}]},{"name":"bar","tests":[{"name":"bar","duration":2,"state":"failed","error":{}}],"hooks":[{"duration":1,"title":"before each","state":"passed"}]},{"name":"foo","tests":[{"name":"foo","duration":1,"state":"passed"}],"hooks":[{"duration":1,"title":"before each","state":"failed"}]},{"name":"bar","tests":[{"name":"bar","duration":2,"state":"failed","error":{}}],"hooks":[{"duration":1,"title":"before each","state":"passed"}]}],"specs":["foo.js","bar.js"],"state":{"passed":2,"failed":2,"skipped":0}}",
  ],
]
`;
